internal/controller/apiextensions/composite/environment/selector.go

```diff
@@ -132,7 +135,7 @@ func (s *APIEnvironmentSelector) lookUpConfigs(ctx context.Context, cr resource.
 func (s *APIEnvironmentSelector) buildEnvironmentConfigRefFromSelector(cl *v1alpha1.EnvironmentConfigList, selector *v1.EnvironmentSourceSelector) ([]corev1.ObjectReference, error) {
 	ec := make([]v1alpha1.EnvironmentConfig, 0)
 
-	if len(cl.Items) == 0 {
+	if cl == nil || len(cl.Items) == 0 {
```

Comment:
should be a regular case to pass `nil` for `cl` in this function? `lookUpConfig` never returns `nil`, `nil` so perhaps we should return error instead? Otherwise, could we add selector unit test for this change?

Reply:
It should never happen but I feel it's ok to handle it as if it had 0 items.

Reply:
If it should never happen, then passing `nil` here might smell like a bug, and failing early would uncover it. 

https://en.wikipedia.org/wiki/Design_by_contract

Reply:
Then let me rephrase: "it can happen and should be handled as if it was an empty list"

Reply:
ok, can you please add selector unit test for that situation?